tst/EnergyPlus/unit/SimAirServingZones.unit.cc

```diff
@@ -630,4 +635,938 @@ TEST_F(EnergyPlusFixture, GetAirPathData_ControllerLockout2)
+        "    ReturnAirPath2,          !- Name",
+        "    Demand 2 Out Node,         !- Return Air Path Outlet Node Name",
+        "    AirLoopHVAC:ZoneMixer,   !- Component 1 Object Type",
+        "    Zone Return Air Mixer 2;   !- Component 1 Name",
+
+        "  AirLoopHVAC:ZoneMixer,",
+        "    Zone Return Air Mixer 2,   !- Name",
+        "    Demand 2 Out Node,         !- Outlet Node Name",
+        "    Space2 Ret Node;         !- Inlet 2 Node Name",
+
+        "  Fan:VariableVolume,",
+        "    Supply Fan 2,            !- Name",
+        "    ,                        !- Availability Schedule Name",
+        "    0.7,                     !- Fan Total Efficiency",
+        "    600.0,                   !- Pressure Rise {Pa}",
+        "    1.0,                     !- Maximum Flow Rate {m3/s}",
+        "    FixedFlowRate,           !- Fan Power Minimum Flow Rate Input Method",
+        "    ,                        !- Fan Power Minimum Flow Fraction",
+        "    0.35326,                 !- Fan Power Minimum Air Flow Rate {m3/s}",
+        "    0.9,                     !- Motor Efficiency",
+        "    1.0,                     !- Motor In Airstream Fraction",
+        "    0.0015302446,            !- Fan Power Coefficient 1",
+        "    0.0052080574,            !- Fan Power Coefficient 2",
+        "    1.1086242,               !- Fan Power Coefficient 3",
+        "    -0.11635563,             !- Fan Power Coefficient 4",
+        "    0.000,                   !- Fan Power Coefficient 5",
+        "    VAV Sys 2 Inlet Node,    !- Air Inlet Node Name",
+        "    VAV Sys 2 Outlet Node;   !- Air Outlet Node Name",
+
+    });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+    bool ErrorsFound = false;
+    HeatBalanceManager::GetZoneData(ErrorsFound);
+    ASSERT_FALSE(ErrorsFound);
+    EXPECT_TRUE(compare_err_stream(""));
+    DataZoneEquipment::GetZoneEquipmentData1();
+    EXPECT_TRUE(compare_err_stream(""));
+    ASSERT_FALSE(ErrorsFound);
+    ZoneAirLoopEquipmentManager::GetZoneAirLoopEquipment();
+    EXPECT_TRUE(compare_err_stream(""));
+    ASSERT_FALSE(ErrorsFound);
+    SingleDuct::GetSysInput();
+    EXPECT_TRUE(compare_err_stream(""));
+    ASSERT_FALSE(ErrorsFound);
+    SplitterComponent::GetSplitterInput();
+    EXPECT_TRUE(compare_err_stream(""));
+    SimAirServingZones::GetAirPathData();
+    // Expect warnings about no controllers, clear err_stream
+    EXPECT_TRUE(has_err_output(true));
+    SimAirServingZones::InitAirLoops(true);
+    EXPECT_TRUE(compare_err_stream(""));
+    ASSERT_FALSE(ErrorsFound);
+    // And finally, all of this gymnastics just to check if the airloopnums get set correctly
+    // For this test, each ADUs should be connected to a different airloop
+    EXPECT_EQ(DataZoneEquipment::ZoneEquipConfig(1).InletNodeAirLoopNum(1), 1);
+    EXPECT_EQ(DataZoneEquipment::ZoneEquipConfig(2).InletNodeAirLoopNum(1), 2);
+}
+
+TEST_F(EnergyPlusFixture, InitAirLoops_2AirLoop3ADUa)
```

Comment:
This one adds a 3rd ADU, so zone 2 has 2 ADUs, airloop 1 serves 2 ADUs in different zones, airloop 2 serves the other ADU in zone 2, this one will pass as-is.